sstables/generation_type.hh

```diff
@@ -22,7 +22,9 @@ public:
 private:
     value_type _value;
 public:
-    generation_type() = delete;
+    generation_type() noexcept
+        : _value(0)
+    {}
```

Comment:
Seems wrong. Why would we want the default constructor to generate a random ID? Just like the default constructor for int64_t doesn't generate a random int.

Creating a new generation should be explicit.

Reply:
I agree, this is too subtle. Our UUID:s also work this way: default ctor creates null UUID, and we have a factory method to generate a random one.

Reply:
we need a default ctor so we are able to accept `generation_type` as an boost::program::option, which uses `boost::lexical_cast<T>()` to cast a `std::string` to the given type. see [`tools/scylla-sstable.cc`](https://github.com/scylladb/scylladb/blob/fb898d214c5aa36689c941b015466a662471f52e/tools/scylla-sstable.cc#L2468) after Benny's change of https://github.com/scylladb/scylladb/pull/12994/commits/5764aed61ad9438201ac4cfd0fc73c9a17ac50a6#diff-4cd1a5351a4e8c9836e4cdf6eaa67498e53d365fe9ab76dcae4baa9e6ac4cdc0R2469. despite that this change has not been landed. i think the change makes sense in general. 

probably once we have UUID based generation id, we will need to use `std::variant<std::monostate, int64_t, UUID>`. otherwise we should have `std::variant<UUID, int64_t>`. as, at least, we can tell if the uuid is generated with the value-initialized ctor just like `UUID{}`. 

or we will need to provide a customized validator for this option.

Reply:
I'm not saying we shouldn't have a default constructor, just that it should create a "null" UUID, not a random one.

Reply:
ahh, i see. it makes more sense.